server/src/main/java/org/elasticsearch/search/aggregations/support/BuiltinValuesSourceType.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.support;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.index.fielddata.IndexFieldData;
+import org.elasticsearch.index.fielddata.IndexGeoPointFieldData;
+import org.elasticsearch.index.fielddata.IndexNumericFieldData;
+import org.elasticsearch.index.fielddata.IndexOrdinalsFieldData;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.RangeFieldMapper;
+import org.elasticsearch.script.AggregationScript;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.aggregations.AggregationExecutionException;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.function.LongSupplier;
+
+public enum BuiltinValuesSourceType implements Writeable, ValuesSourceType {
+    ANY {
+        @Override
+        public ValuesSource getEmpty() {
+            // TODO: Implement this or get rid of ANY
+            throw new UnsupportedOperationException("ValuesSourceType.ANY is still a special case");
```

Comment:
Should be `BuiltinValuesSourceType.ANY` right?

Not sure about the error message, but don't have a better suggestion at the moment :/

Reply:
Yeah, this was my main hesitation on merging this to master.  ANY is still handled in the "old style" with special cases scattered through `ValuesSourceConfig`.  I'm pretty sure we'll never hit this error message, but given the complexity it's hard to be absolutely sure.